src/sidebar/components/HypothesisApp.tsx

```diff
@@ -69,6 +69,14 @@ function HypothesisApp({
   const showShareButton =
     !isThirdParty || exportAnnotations || importAnnotations;
 
+  const roundedCornersEnabled = store.isFeatureEnabled('rounded_corners');
+  useEffect(() => {
+    // Setting this attribute will enable the appropriate CSS variables
+    document
+      .querySelector('html')
+      ?.toggleAttribute('data-rounded-corners', roundedCornersEnabled);
+  }, [roundedCornersEnabled]);
```

Comment:
I'm thinking now if should actually be `useLayoutEffect` 🤔

Reply:
Yes. `useLayoutEffect` runs synchronously at the end of a render. `useEffect` runs after the next repaint. Updates that have an immediate visual effect are best done with `useLayoutEffect` to avoid a flash and wasted paint.

Reply:
Actually no. I think `useEffect` makes sense here, because any UI change happening as a side effect of this is natively handled by the browser.

Reply:
As an alternative to using an effect, you could also just modify the CSS so that `data-rounded-corners` is set on a root element in the app and not the `<html>` element. The CSS var changes will still propagate down to the children.

Reply:
> As an alternative to using an effect, you could also just modify the CSS so that data-rounded-corners is set on a root element in the app and not the <html> element. The CSS var changes will still propagate down to the children.

I'm not sure I follow 😅

Reply:
Change the CSS rule from `html[data-rounded-corners]` to `[data-rounded-corners]`, then set the `data-rounded-corners` on the root element rendered by this component. Then no effects are required and everything can be set in the initial render.

Reply:
You could even just set the properties via inline styles.